src/Model/Item.php

```diff
@@ -1453,6 +1453,16 @@ public static function insert($item, $force_parent = false, $notify = false, $do
 			return 0;
 		}
 
+		if (!empty($uid) && Contact::isBlockedByUser($item['author-link'], $uid)) {
+			Logger::notice('Author is blocked by user', ['author-link' => $item['author-link'], 'uid' => $uid, 'item-uri' => $item['uri']]);
+			return 0;
+		}
+
+		if (!empty($uid) && ($item['parent-uri'] === $item['uri']) && Contact::isIgnoredByUser($item['author-link'], $uid)) {
```

Comment:
Ignore only affects network stream display, not item delivery. See https://github.com/friendica/friendica/issues/6897#issuecomment-491582525 (I think)

Reply:
No. "Ignoring" does mean that you don't want to read starting posts from that contact, but this contact should be able to comment and should be able to receive your posts. And this is done on this level. (See the "readonly" field in the contact)

Reply:
What I mean is that I should be able to choose to display single top-level posts by an account I'm following but ignoring, even (and especially) if it isn't a public post. This includes the `/display` page, of course, but also the `/contact/xxx/conversation` page for example.

Reply:
I must confess that I'm currently confused about how it is supposed to work. But when it is done the way it seems that it had been done, then we don't have a chance to suppress stuff that is reshared by an ignored contact.

Reply:
For ignored accounts, it should just not show up in streams, like `/network` or `/community`. Searches, contact conversations and `/display` should still display the top-level and reshared posts.

So the check for blocked accounts is right in its place when we store the item, but the ignore check should take place at the stream level. Or the ignore check when we store the item should set a flag (in `user-item`, tentatively) instead of discarding the whole item.